packages/clerk-js/src/ui/signUp/SignUpContinue.tsx

```diff
@@ -0,0 +1,350 @@
+  if (!signUp.id) {
+    void navigate(displayConfig.signUpUrl);
+    return null;
+  }
+
+  // Pre-populate fields from existing sign-up object
+  const formFields = {
+    firstName: useFieldState('first_name', signUp.firstName || ''),
+    lastName: useFieldState('last_name', signUp.lastName || ''),
+    emailAddress: useFieldState('email_address', signUp.emailAddress || ''),
+    username: useFieldState('username', signUp.username || ''),
+    phoneNumber: useFieldState('phone_number', signUp.phoneNumber || ''),
+    password: useFieldState('password', ''),
+  } as const;
+
+  type FormFieldsKey = keyof typeof formFields;
+
+  const [error, setError] = React.useState<string | undefined>();
+
+  const hasVerifiedEmail = signUp.verifications?.emailAddress?.status == 'verified';
+  const hasVerifiedPhone = signUp.verifications?.phoneNumber?.status == 'verified';
+  const hasVerifiedExternalAccount = signUp.verifications?.externalAccount?.status == 'verified';
+
+  const fields = determineFirstPartyFields(environment, false);
+
+  // Show only fields absolutely necessary for completing sign-up to minimize fiction
+
+  if (hasVerifiedEmail) {
+    delete fields.emailAddress;
+    delete fields.emailOrPhone;
+  }
+
+  if (hasVerifiedPhone) {
+    delete fields.phoneNumber;
+    delete fields.emailOrPhone;
+  }
+
+  if (hasVerifiedExternalAccount) {
+    delete fields.password;
+  }
+
+  if (signUp.firstName) {
+    delete fields.firstName;
+  }
+
+  if (signUp.lastName) {
+    delete fields.lastName;
+  }
+
+  if (signUp.username) {
+    delete fields.username;
+  }
+
+  // Remove any non-required fields
+  Object.entries(fields).forEach(([k, v]) => {
+    if (v !== 'required') {
+      // @ts-ignore
+      delete fields[k];
+    }
+  });
+
+  const oauthOptions = userSettings.socialProviderStrategies;
+  const web3Options = userSettings.web3FirstFactors;
+
+  // Handle oauth errors?
+
+  // React.useEffect(() => {
+  //   async function handleOauthError() {
+  //     const error = signUp.verifications.externalAccount.error;
+  //
+  //     if (error) {
+  //       switch (error.code) {
+  //         case ERROR_CODES.NOT_ALLOWED_TO_SIGN_UP:
+  //         case ERROR_CODES.OAUTH_ACCESS_DENIED:
+  //           setError(error.longMessage);
+  //           break;
+  //         default:
+  //           // Error from server may be too much information for the end user, so set a generic error
+  //           setError('Unable to complete action at this time. If the problem persists please contact support.');
+  //       }
+  //
+  //       // TODO: This is a hack to reset the sign in attempt so that the oauth error
+  //       // does not persist on full page reloads.
+  //       //
+  //       // We will revise this strategy as part of the Clerk DX epic.
+  //       void (await signUp.create({}));
+  //     }
+  //   }
+  //
+  //   void handleOauthError();
+  // });
+
+  const handleChangeActive = (type: ActiveIdentifier) => (e: React.MouseEvent) => {
```

Comment:
⛔  There is a lot of code duplication between the `<SignUpStart/>` and the `<SignUpContinue/>` components that  increases the maintenance overhead. Let's leverage composition and make them both depend on a new, common `<SignUpForm/>` component.